gui/login/static/LoginController.js

```diff
@@ -26,6 +26,26 @@ angular
   
   $scope.$watch(function() { return Project.project; }, function(newValue, oldValue) { $scope.project = newValue; });
   
+  $scope.enableOAuth = SigninUrl.indexOf("/oauth2/") !== -1;
+  $scope.signinLink = $scope.enableOAuth ? SigninUrl : '/login/' + ($rootScope.rd ? 'rd=' : '');
+  
+  // User should not be here if OAuth is enabled...
+  // Navigate them to the correct place
+  if ($scope.enableOAuth) {
+    $window.location.href = $scope.signinLink;
+    return;
+  };
+  
+  $scope.clickCreateAccountLink = function() {
+    if ($scope.enableOAuth) { return false; }
+    $window.location.href = '/login/register' + ($rootScope.rd ? '?rd=' + $rootScope.rd : '')
+  };
+  
+  $scope.clickForgotPasswordLink = function() {
+    if ($scope.enableOAuth) { return false; }
+    $window.location.href = '/login/recover' + ($rootScope.rd ? '?rd=' + $rootScope.rd : '')
+  };
+  
```

Comment:
Detect OAuth configuration in LoginPage... Redirect if odd state encountered, disable page in case redirect doesn't work.